src/invert.ts

```diff
@@ -0,0 +1,46 @@
+import { purry } from './purry';
+
+type Invertable = Record<PropertyKey, PropertyKey> | Array<PropertyKey>;
```

Comment:
You don't actually need invertable, you can infer the types in the output

Reply:
I'm happy to cut it, but don't I need `Record<PropertyKey, PropertyKey>` somewhere to indicate that the values are themselves valid keys? Or am I missing something?

Reply:
I'm happy to cut it, but don't I then need ` T extends Record<PropertyKey, PropertyKey>` to ensure a caller of the function doesn't provide an object whose values aren't valid property keys?

I've left it for now, but please let me know what you'd like me to do as an alternative and I'll drop it. 🙂